homeassistant/components/gardena_bluetooth/config_flow.py

```diff
@@ -0,0 +1,144 @@
+"""Config flow for Gardena Bluetooth integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from bleak import BleakClient
+from bleak.exc import BleakError
+from gardena_bluetooth import CharacteristicNotFound, read_char
+from gardena_bluetooth.const import DeviceInformation, ScanService
+from gardena_bluetooth.parse import ManufacturerData, ProductGroup
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfo,
+    async_ble_device_from_address,
+    async_discovered_service_info,
+)
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _is_supported(discovery_info: BluetoothServiceInfo):
+    """Check if device is supported."""
+    if ScanService not in discovery_info.service_uuids:
+        return False
+
+    if not (data := discovery_info.manufacturer_data.get(ManufacturerData.company)):
+        _LOGGER.debug("Missing manufacturer data: %s", discovery_info)
+        return False
+
+    manufacturer_data = ManufacturerData.decode(data)
+    if manufacturer_data.group != ProductGroup.WATER_CONTROL:
+        _LOGGER.debug("Unsupported device: %s", manufacturer_data)
+        return False
+
+    return True
+
+
+def _get_name(discovery_info: BluetoothServiceInfo):
+    if discovery_info.name and discovery_info.name != discovery_info.address:
+        return discovery_info.name
+    return "Gardena Device"
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Gardena Bluetooth."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.devices: dict[str, str] = {}
+        self.address: str | None
+
+    async def async_read_data(self):
+        """Try to connect to device and extract information."""
+        assert self.address
+        ble_device = async_ble_device_from_address(
+            self.hass, self.address, connectable=True
+        )
+        assert ble_device
+
+        try:
+            async with BleakClient(ble_device) as client:
+                model = await read_char(client, DeviceInformation.model_number)
+                _LOGGER.debug("Found device with model: %s", model)
```

Comment:
Same comment as the `get` here.  We don't have any other integrations besides the bluetooth le tracker that create `BleakClient` directly.

Reply:
I've wrapped up the client. Suppose it meant i could contain bleak exceptions better, so that was a plus.